### ISSUE TITLE
Failed to start project 

### ISSUE THREAD

**Issue by khoinpb96** [langchain-ai/langgraph-studio#13]
Hi team, after try setup and start my project I got this issue
```
Failed to start project <project-name>
failed to solve: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
```
<img width="715" alt="image" src="https://github.com/user-attachments/assets/3daff273-bb1a-4b10-bc7d-57a31a499564">

**Comment by sprour** [comment]
Check this thread https://github.com/langchain-ai/langgraph-studio/issues/5

**Comment by khoinpb96** [comment]
> Check this thread #5
Thanks but I still got the issue

Here's my langgraph.json 
```json
{
  "dependencies": ["."],
  "graphs": {
    "chat_graph": "./app/graphs/chat_graph.py:chat_graph"
  },
  "env": "./.env"
}

```

docker-compose version is `2.27.0`

Here's the log
```
#7 [langgraph-api 3/4] RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*
#7 1.051 Obtaining file:///deps/maidas
#7 1.052   Installing build dependencies: started
#7 2.960   Installing build dependencies: finished with status 'done'
#7 2.962   Checking if build backend supports build_editable: started
#7 3.146   Checking if build backend supports build_editable: finished with status 'done'
#7 3.147   Getting requirements to build editable: started
#7 3.332   Getting requirements to build editable: finished with status 'done'
#7 3.333   Preparing editable metadata (pyproject.toml): started
#7 3.641   Preparing editable metadata (pyproject.toml): finished with status 'error'
#7 3.645   error: subprocess-exited-with-error
#7 3.645   
#7 3.645   × Preparing editable metadata (pyproject.toml) did not run successfully.
#7 3.645   │ exit code: 1
#7 3.645   ╰─> [17 lines of output]
#7 3.645       Traceback (most recent call last):
#7 3.645         File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
#7 3.645           main()
#7 3.645         File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
#7 3.645           json_out['return_val'] = hook(**hook_input['kwargs'])
#7 3.645                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#7 3.645         File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 181, in prepare_metadata_for_build_editable
#7 3.645           return hook(metadata_directory, config_settings)
#7 3.645                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#7 3.645         File "/tmp/pip-build-env-ydbw3h_g/overlay/lib/python3.11/site-packages/poetry/core/masonry/api.py", line 43, in prepare_metadata_for_build_wheel
#7 3.645           builder = WheelBuilder(poetry)
#7 3.645                     ^^^^^^^^^^^^^^^^^^^^
#7 3.645         File "/tmp/pip-build-env-ydbw3h_g/overlay/lib/python3.11/site-packages/poetry/core/masonry/builders/wheel.py", line 62, in __init__
#7 3.645           super().__init__(poetry, executable=executable)
#7 3.645         File "/tmp/pip-build-env-ydbw3h_g/overlay/lib/python3.11/site-packages/poetry/core/masonry/builders/builder.py", line 45, in __init__
#7 3.645           raise RuntimeError(
#7 3.645       RuntimeError: Building a package is not possible in non-package mode.
#7 3.645       [end of output]
#7 3.645   
#7 3.645   note: This error originates from a subprocess, and is likely not a problem with pip.
#7 3.646 error: metadata-generation-failed
#7 3.646 
#7 3.646 × Encountered error while generating package metadata.
#7 3.646 ╰─> See above for output.
#7 3.646 
#7 3.646 note: This is an issue with the package mentioned above, not pip.
#7 3.646 hint: See above for details.
#7 3.838 
#7 3.838 [notice] A new release of pip is available: 24.0 -> 24.2
#7 3.838 [notice] To update, run: pip install --upgrade pip
#7 ERROR: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
------
 > [langgraph-api 3/4] RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*:
3.646 error: metadata-generation-failed
3.646 
3.646 × Encountered error while generating package metadata.
3.646 ╰─> See above for output.
3.646 
3.646 note: This is an issue with the package mentioned above, not pip.
3.646 hint: See above for details.
3.838 
3.838 [notice] A new release of pip is available: 24.0 -> 24.2
3.838 [notice] To update, run: pip install --upgrade pip
------
failed to solve: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
```

**Comment by hwchase17** [comment]
It looks like you are trying to build with pyproject.toml that has package-mode=false - we expect it be true

https://langchain-ai.github.io/langgraph/cloud/deployment/setup_pyproject/

Is that correct/does changing that fix it?

**Comment by khoinpb96** [comment]
Still got this issue but the log's now different

```
Starting LangGraph API server...
For local dev, requires env var LANGSMITH_API_KEY with access to LangGraph Cloud closed beta.
For production use, requires a license key in env var LANGGRAPH_CLOUD_LICENSE_KEY.
/ Building...#0 building with "colima" instance using docker driver

#1 [langgraph-api internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [langgraph-api internal] load build definition from Dockerfile
#2 transferring dockerfile: 338B done
#2 DONE 0.0s

#3 [langgraph-api internal] load metadata for docker.io/langchain/langgraph-api:3.11
#3 DONE 0.0s

#4 [langgraph-api 1/4] FROM docker.io/langchain/langgraph-api:3.11
#4 DONE 0.0s

#5 [langgraph-api internal] load build context
#5 transferring context: 200.35kB 0.2s done
#5 DONE 0.2s

#6 [langgraph-api 2/4] ADD . /deps/xxx
#6 CACHED

#7 [langgraph-api 3/4] RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*
#7 1.033 Obtaining file:///deps/xxx
#7 1.035   Installing build dependencies: started
#7 3.057   Installing build dependencies: finished with status 'done'
#7 3.058   Checking if build backend supports build_editable: started
#7 3.249   Checking if build backend supports build_editable: finished with status 'done'
#7 3.250   Getting requirements to build editable: started
#7 3.439   Getting requirements to build editable: finished with status 'done'
#7 3.440   Preparing editable metadata (pyproject.toml): started
#7 3.753   Preparing editable metadata (pyproject.toml): finished with status 'done'
#7 3.911 Collecting authlib<2.0.0,>=1.3.0 (from app==0.0.1)
#7 4.092   Downloading Authlib-1.3.1-py2.py3-none-any.whl.metadata (3.8 kB)
#7 4.171 Collecting datadog<0.50.0,>=0.49.1 (from app==0.0.1)
#7 4.230   Downloading datadog-0.49.1-py2.py3-none-any.whl.metadata (10 kB)
#7 4.363 Collecting fastapi[all] (from app==0.0.1)
#7 4.420   Downloading fastapi-0.112.0-py3-none-any.whl.metadata (27 kB)
#7 4.507 Collecting gunicorn (from app==0.0.1)
#7 4.565   Downloading gunicorn-22.0.0-py3-none-any.whl.metadata (4.4 kB)
#7 4.628 Collecting httpx-sse<0.5.0,>=0.4.0 (from app==0.0.1)
#7 4.690   Downloading httpx_sse-0.4.0-py3-none-any.whl.metadata (9.0 kB)
#7 4.777 Collecting inflect<8.0.0,>=7.3.1 (from app==0.0.1)
#7 4.833   Downloading inflect-7.3.1-py3-none-any.whl.metadata (21 kB)
#7 4.978 Collecting langchain<0.3.0,>=0.2.7 (from app==0.0.1)
#7 5.025   Downloading langchain-0.2.12-py3-none-any.whl.metadata (7.1 kB)
#7 5.084 Collecting langchain-community<0.3.0,>=0.2.1 (from app==0.0.1)
#7 5.137   Downloading langchain_community-0.2.11-py3-none-any.whl.metadata (2.7 kB)
#7 5.203 Collecting langchain_openai==0.1.7 (from app==0.0.1)
#7 5.258   Downloading langchain_openai-0.1.7-py3-none-any.whl.metadata (2.5 kB)
#7 5.325 Collecting langchain_postgres==0.0.9 (from app==0.0.1)
#7 5.388   Downloading langchain_postgres-0.0.9-py3-none-any.whl.metadata (4.0 kB)
#7 5.462 Collecting langchainhub<0.2.0,>=0.1.17 (from app==0.0.1)
#7 5.531   Downloading langchainhub-0.1.20-py3-none-any.whl.metadata (659 bytes)
#7 5.534 Requirement already satisfied: langgraph<0.2.0,>=0.1.11 in /usr/local/lib/python3.11/site-packages (from app==0.0.1) (0.1.19)
#7 5.582 Collecting langserve<0.3.0,>=0.2.0 (from app==0.0.1)
#7 5.645   Downloading langserve-0.2.2-py3-none-any.whl.metadata (39 kB)
#7 5.668 Requirement already satisfied: langsmith<0.2.0,>=0.1.62 in /usr/local/lib/python3.11/site-packages (from app==0.0.1) (0.1.96)
#7 5.742 Collecting opentelemetry-distro<0.46,>=0.45b0 (from app==0.0.1)
#7 5.804   Downloading opentelemetry_distro-0.45b0-py3-none-any.whl.metadata (1.4 kB)
#7 5.872 Collecting opentelemetry-exporter-otlp<2.0.0,>=1.24.0 (from app==0.0.1)
#7 5.943   Downloading opentelemetry_exporter_otlp-1.26.0-py3-none-any.whl.metadata (2.3 kB)
#7 6.028 Collecting opentelemetry-instrumentation-fastapi<0.46,>=0.45b0 (from app==0.0.1)
#7 6.094   Downloading opentelemetry_instrumentation_fastapi-0.45b0-py3-none-any.whl.metadata (2.0 kB)
#7 6.242 Collecting pandas<3.0.0,>=2.2.2 (from app==0.0.1)
#7 6.313   Downloading pandas-2.2.2-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata (19 kB)
#7 6.360 Collecting pgvector<0.3.0,>=0.2.5 (from app==0.0.1)
#7 6.421   Downloading pgvector-0.2.5-py2.py3-none-any.whl.metadata (9.9 kB)
#7 6.532 Collecting psycopg2<3.0.0,>=2.9.9 (from app==0.0.1)
#7 6.587   Downloading psycopg2-2.9.9.tar.gz (384 kB)
#7 6.742      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 384.9/384.9 kB 2.6 MB/s eta 0:00:00
#7 6.793   Preparing metadata (setup.py): started
#7 7.248   Preparing metadata (setup.py): finished with status 'error'
#7 7.251   error: subprocess-exited-with-error
#7 7.251   
#7 7.251   × python setup.py egg_info did not run successfully.
#7 7.251   │ exit code: 1
#7 7.251   ╰─> [23 lines of output]
#7 7.251       running egg_info
#7 7.251       creating /tmp/pip-pip-egg-info-dx0ln9oh/psycopg2.egg-info
#7 7.251       writing /tmp/pip-pip-egg-info-dx0ln9oh/psycopg2.egg-info/PKG-INFO
#7 7.251       writing dependency_links to /tmp/pip-pip-egg-info-dx0ln9oh/psycopg2.egg-info/dependency_links.txt
#7 7.251       writing top-level names to /tmp/pip-pip-egg-info-dx0ln9oh/psycopg2.egg-info/top_level.txt
#7 7.251       writing manifest file '/tmp/pip-pip-egg-info-dx0ln9oh/psycopg2.egg-info/SOURCES.txt'
#7 7.251       
#7 7.251       Error: pg_config executable not found.
#7 7.251       
#7 7.251       pg_config is required to build psycopg2 from source.  Please add the directory
#7 7.251       containing pg_config to the $PATH or specify the full executable path with the
#7 7.251       option:
#7 7.251       
#7 7.251           python setup.py build_ext --pg-config /path/to/pg_config build ...
#7 7.251       
#7 7.251       or with the pg_config option in 'setup.cfg'.
#7 7.251       
#7 7.251       If you prefer to avoid building psycopg2 from source, please install the PyPI
#7 7.251       'psycopg2-binary' package instead.
#7 7.251       
#7 7.251       For further information please check the 'doc/src/install.rst' file (also at
#7 7.251       <https://www.psycopg.org/docs/install.html>).
#7 7.251       
#7 7.251       [end of output]
#7 7.251   
#7 7.251   note: This error originates from a subprocess, and is likely not a problem with pip.
#7 7.252 error: metadata-generation-failed
#7 7.252 
#7 7.252 × Encountered error while generating package metadata.
#7 7.252 ╰─> See above for output.
#7 7.252 
#7 7.252 note: This is an issue with the package mentioned above, not pip.
#7 7.252 hint: See above for details.
#7 7.447 
#7 7.447 [notice] A new release of pip is available: 24.0 -> 24.2
#7 7.447 [notice] To update, run: pip install --upgrade pip
#7 ERROR: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
------
 > [langgraph-api 3/4] RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*:
7.252 error: metadata-generation-failed
7.252 
7.252 × Encountered error while generating package metadata.
7.252 ╰─> See above for output.
7.252 
7.252 note: This is an issue with the package mentioned above, not pip.
7.252 hint: See above for details.
7.447 
7.447 [notice] A new release of pip is available: 24.0 -> 24.2
7.447 [notice] To update, run: pip install --upgrade pip
------
failed to solve: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
```

Don't know what leds to `Error: pg_config executable not found.` but I've already installed `postgresql`

**Comment by nfcampos** [comment]
Looks like you are trying to install psycopg2, without having postgres installed. There are two ways to fix this
- switch your dependency from psycopg2 to psycopg2-binary
- install postgres inside the docker image by making use of dockerfile_lines in langgraph.json
Id recommend the first option as it will result in a smaller docker image

**Comment by pru007** [comment]
Hi, I am getting similar issue.

```

#7 14.99 Building wheels for collected packages: app, PyStemmer
#7 14.99   Building editable for app (pyproject.toml): started
#7 15.24   Building editable for app (pyproject.toml): finished with status 'error'
#7 15.24   error: subprocess-exited-with-error
#7 15.24   
#7 15.24   × Building editable for app (pyproject.toml) did not run successfully.
#7 15.24   │ exit code: 1
#7 15.24   ╰─> [28 lines of output]
#7 15.24       Traceback (most recent call last):
#7 15.24         File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
#7 15.24           main()
#7 15.24         File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
#7 15.24           json_out['return_val'] = hook(**hook_input['kwargs'])
#7 15.24                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#7 15.24         File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 273, in build_editable
#7 15.24           return hook(wheel_directory, config_settings, metadata_directory)
#7 15.24                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#7 15.24         File "/tmp/pip-build-env-6s4c_bda/overlay/lib/python3.11/site-packages/poetry/core/masonry/api.py", line 82, in build_editable
#7 15.24           return WheelBuilder.make_in(
#7 15.24                  ^^^^^^^^^^^^^^^^^^^^^
#7 15.24         File "/tmp/pip-build-env-6s4c_bda/overlay/lib/python3.11/site-packages/poetry/core/masonry/builders/wheel.py", line 88, in make_in
#7 15.24           wb.build(target_dir=directory)
#7 15.24         File "/tmp/pip-build-env-6s4c_bda/overlay/lib/python3.11/site-packages/poetry/core/masonry/builders/wheel.py", line 118, in build
#7 15.24           self._add_pth(zip_file)
#7 15.24         File "/tmp/pip-build-env-6s4c_bda/overlay/lib/python3.11/site-packages/poetry/core/masonry/builders/wheel.py", line 147, in _add_pth
#7 15.24           for include in self._module.includes:
#7 15.24                          ^^^^^^^^^^^^
#7 15.24         File "/usr/local/lib/python3.11/functools.py", line 1001, in __get__
#7 15.24           val = self.func(instance)
#7 15.24                 ^^^^^^^^^^^^^^^^^^^
#7 15.24         File "/tmp/pip-build-env-6s4c_bda/overlay/lib/python3.11/site-packages/poetry/core/masonry/builders/builder.py", line 97, in _module
#7 15.24           return Module(
#7 15.24                  ^^^^^^^
#7 15.24         File "/tmp/pip-build-env-6s4c_bda/overlay/lib/python3.11/site-packages/poetry/core/masonry/utils/module.py", line 65, in __init__
#7 15.24           raise ModuleOrPackageNotFound(
#7 15.24       poetry.core.masonry.utils.module.ModuleOrPackageNotFound: No file/folder found for package app
#7 15.24       [end of output]
#7 15.24   
#7 15.24   note: This error originates from a subprocess, and is likely not a problem with pip.
#7 15.24   ERROR: Failed building editable for app
#7 15.25   Building wheel for PyStemmer (setup.py): started
#7 16.08   Building wheel for PyStemmer (setup.py): finished with status 'error'
#7 16.08   error: subprocess-exited-with-error
#7 16.08   
#7 16.08   × python setup.py bdist_wheel did not run successfully.
#7 16.08   │ exit code: 1
#7 16.08   ╰─> [19 lines of output]
#7 16.08       /usr/local/lib/python3.11/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
#7 16.08         warnings.warn(
#7 16.08       running bdist_wheel
#7 16.08       running build
#7 16.08       running build_ext
#7 16.08       Compiling src/Stemmer.pyx because it changed.
#7 16.08       [1/1] Cythonizing src/Stemmer.pyx
#7 16.08       /tmp/pip-install-6vos5_js/pystemmer_9e644c14aa544183847e28d9c017e44b/.eggs/Cython-3.0.11-py3.11-linux-aarch64.egg/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-6vos5_js/pystemmer_9e644c14aa544183847e28d9c017e44b/src/Stemmer.pyx
#7 16.08         tree = Parsing.p_module(s, pxd, full_module_name)
#7 16.08       building 'Stemmer' extension
#7 16.08       creating build
#7 16.08       creating build/temp.linux-aarch64-cpython-311
#7 16.08       creating build/temp.linux-aarch64-cpython-311/libstemmer_c-2.2.0
#7 16.08       creating build/temp.linux-aarch64-cpython-311/libstemmer_c-2.2.0/libstemmer
#7 16.08       creating build/temp.linux-aarch64-cpython-311/libstemmer_c-2.2.0/runtime
#7 16.08       creating build/temp.linux-aarch64-cpython-311/libstemmer_c-2.2.0/src_c
#7 16.08       creating build/temp.linux-aarch64-cpython-311/src
#7 16.08       gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Ilibstemmer_c-2.2.0/include -I/usr/local/include/python3.11 -c libstemmer_c-2.2.0/libstemmer/libstemmer_utf8.c -o build/temp.linux-aarch64-cpython-311/libstemmer_c-2.2.0/libstemmer/libstemmer_utf8.o
#7 16.08       error: command 'gcc' failed: No such file or directory
#7 16.08       [end of output]
#7 16.08   
#7 16.08   note: This error originates from a subprocess, and is likely not a problem with pip.
#7 16.08   ERROR: Failed building wheel for PyStemmer
#7 16.08   Running setup.py clean for PyStemmer
#7 16.46 Failed to build app PyStemmer
#7 16.46 ERROR: Could not build wheels for app, PyStemmer, which is required to install pyproject.toml-based projects
#7 16.46 
#7 16.46 [notice] A new release of pip is available: 24.0 -> 24.2
#7 16.46 [notice] To update, run: pip install --upgrade pip
#7 ERROR: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
------
 > [langgraph-api stage-0 3/4] RUN --mount=type=cache,target=/root/.cache/pip PYTHONDONTWRITEBYTECODE=1 pip install -c /api/constraints.txt -e /deps/*:
16.08       [end of output]
16.08   
16.08   note: This error originates from a subprocess, and is likely not a problem with pip.
16.08   ERROR: Failed building wheel for PyStemmer
16.08   Running setup.py clean for PyStemmer
16.46 Failed to build app PyStemmer
16.46 ERROR: Could not build wheels for app, PyStemmer, which is required to install pyproject.toml-based projects
16.46 
16.46 [notice] A new release of pip is available: 24.0 -> 24.2
16.46 [notice] To update, run: pip install --upgrade pip
------
failed to solve: process "/bin/sh -c PYTHONDONTWRITEBYTECODE=1 pip install -c /api/constraints.txt -e /deps/*" did not complete successfully: exit code: 1
```
I have set package-mode=true.

-Prudhvi
